gluoncv/model_zoo/mask_score_rcnn/mask_score_rcnn.py

```diff
@@ -0,0 +1,1071 @@
+        Names of categories, its length is ``num_class``.
+    mask_channels : int, default is 256
+        Number of channels in mask prediction
+    rcnn_max_dets : int, default is 1000
+        Number of rois to retain in RCNN.
+        Upper bounded by min of rpn_test_pre_nms and rpn_test_post_nms.
+    rpn_test_pre_nms : int, default is 6000
+        Filter top proposals before NMS in testing of RPN.
+    rpn_test_post_nms : int, default is 1000
+        Return top proposal results after NMS in testing of RPN.
+        Will be set to rpn_test_pre_nms if it is larger than rpn_test_pre_nms.
+    target_roi_scale : int, default 1
+        Ratio of mask output roi / input roi. For model with FPN, this is typically 2.
+    num_fcn_convs : int, default 0
+        number of convolution blocks before deconv layer. For FPN network this is typically 4.
+    """
+
+    def __init__(self, features, top_features, classes, mask_channels=256, rcnn_max_dets=1000,
+                 rpn_test_pre_nms=6000, rpn_test_post_nms=1000, target_roi_scale=1, num_fcn_convs=0,
+                 norm_layer=None, norm_kwargs=None, **kwargs):
+        super(MaskScoreRCNN, self).__init__()
+
+        self.FasterRCNN = FasterRCNN(features, top_features, classes,
+                                     rpn_test_pre_nms=rpn_test_pre_nms,
+                                     rpn_test_post_nms=rpn_test_post_nms,
+                                     additional_output=True, **kwargs)
+
+        self.top_features = self.FasterRCNN.top_features
+        self._roi_mode = self.FasterRCNN._roi_mode
+        self._strides = self.FasterRCNN._strides
+        self._pyramid_roi_feats = self.FasterRCNN._pyramid_roi_feats
+        self.num_stages = self.FasterRCNN.num_stages
+        self._batch_size = self.FasterRCNN._batch_size
+        self._num_sample = self.FasterRCNN._num_sample
+        self.num_class = self.FasterRCNN.num_class
+        self.ashape = self.FasterRCNN.ashape
+        self.rpn = self.FasterRCNN.rpn
+        self.features = features
+        self._roi_size = kwargs['roi_size']
+        self.short = kwargs['short']
+        self.max_size = kwargs['max_size']
+        self.train_patterns = kwargs['train_patterns']
+
+        if min(rpn_test_pre_nms, rpn_test_post_nms) < rcnn_max_dets:
+            rcnn_max_dets = min(rpn_test_pre_nms, rpn_test_post_nms)
+        self._rcnn_max_dets = rcnn_max_dets
+        with self.name_scope():
+            self.mask = Mask(self._batch_size, classes, mask_channels, num_fcn_convs=num_fcn_convs,
+                             norm_layer=norm_layer, norm_kwargs=norm_kwargs)
+            self.mask_score = MaskScore(self._batch_size, classes,
+                                        mask_channels, rcnn_max_dets=rcnn_max_dets)
+
+            roi_size = (self._roi_size[0] * target_roi_scale, self._roi_size[1] * target_roi_scale)
+            self._target_roi_size = roi_size
+            self.mask_target = MaskTargetGenerator(
```

Comment:
you can put MaskTargetGenerator in a set and invoke in the training loop instead, and you should be able to hybridize the model